code/ui/manager/src/components/preview/FramesRenderer.tsx

```diff
@@ -95,12 +97,12 @@ export const FramesRenderer: FC<FramesRendererProps> = ({
       .reduce((acc, r) => {
         return {
           ...acc,
-          [`storybook-ref-${r.id}`]: `${r.url}/iframe.html?id=${storyId}&viewMode=${viewMode}&refId=${r.id}${stringifiedQueryParams}`,
+          [`storybook-ref-${r.id}`]: `${r.url}/iframe.html?id=${storyIdRef}&viewMode=${viewModeRef}&refId=${r.id}${stringifiedQueryParams}`,
         };
       }, frames);
 
     setFrames(newFrames);
-  }, [storyId, entry, refs]);
+  }, [entry, ...Object.values(refs).map((r) => r.id)]);
```

Comment:
1. If we pass `entry` here, doesn't it rerun everytime you change story (and thus the entry changes)?
2. Do the internals of the refs ever change? I was thinking if we do the filtering outside of the `useEffect` and key on the list of filtered ref ids, we could be resilient to that.